crates/ruff_python_formatter/src/statement/suite.rs

```diff
@@ -323,16 +333,18 @@ impl FormatRule<Suite, PyFormatContext<'_>> for FormatSuite {
     }
 }
 
-/// Returns `true` if a function or class body contains only an ellipsis.
-fn contains_only_an_ellipsis(body: &[Stmt]) -> bool {
+/// Returns `true` if a function or class body contains only an ellipsis with no comments.
+pub(crate) fn contains_only_an_ellipsis(body: &[Stmt], comments: &Comments) -> bool {
     match body {
-        [Stmt::Expr(ast::StmtExpr { value, .. })] => matches!(
-            value.as_ref(),
-            Expr::Constant(ast::ExprConstant {
-                value: Constant::Ellipsis,
-                ..
-            })
-        ),
+        [Stmt::Expr(ast::StmtExpr { value, .. })] if !comments.has_comments(value.as_ref()) => {
+            matches!(
+                value.as_ref(),
+                Expr::Constant(ast::ExprConstant {
+                    value: Constant::Ellipsis,
+                    ..
+                })
+            )
+        }
```

Comment:
I think I was wrong with suggesting `has_comments`. It should be sufficient to these whether `stmt` has any leading comments. Let's move the `has_comments check after the match. Looking up comments is more expensive than testing if it is a const ellipsis expression

```suggestion
        [Stmt::Expr(stmt @ ast::StmtExpr { value, .. })] => {
            matches!(
                value.as_ref(),
                Expr::Constant(ast::ExprConstant {
                    value: Constant::Ellipsis,
                    ..
                })
            ) && !comments.has_leading_comments((stmt)
        }
```